src/DefaultBuilder/src/WebApplicationBuilder.cs

```diff
@@ -212,9 +213,21 @@ public WebApplication Build()
 
             _builtApplication = new WebApplication(_hostBuilder.Build());
 
+            // We know that ConfigurationBuilder constructs ConfigurationRoot with a List<IConfigurationProvider> and all the
+            // IgnoreLoadConfigurationProviders have been "loaded", so we can put the originals back.
+            var providerList = (List<IConfigurationProvider>)((IConfigurationRoot)_builtApplication.Configuration).Providers;
```

Comment:
We should be defensive here. If the user adds a custom `IConfiguration` this will explode. At best it should just noop.